Source/WebCore/Modules/permissions/PermissionStatus.h

```diff
@@ -40,7 +47,7 @@ class ScriptExecutionContext;
 class PermissionStatus final : public PermissionObserver, public ActiveDOMObject, public RefCounted<PermissionStatus>, public EventTarget  {
     WTF_MAKE_ISO_ALLOCATED(PermissionStatus);
 public:
-    static Ref<PermissionStatus> create(ScriptExecutionContext&, PermissionState, const PermissionDescriptor&);
+    static Ref<PermissionStatus> create(ScriptExecutionContext&, PermissionState, const PermissionDescriptor&, PermissionQuerySource, WeakPtr<Page>);
```

Comment:
Let's not pass WeakPtr<Page> by value to avoid refcounting churn, `WeakPtr<Page>&&` is likely what we want here.